### PR TITLE
Update peft dependency, fix initialization and inference with new peft

### PR DESCRIPTION
**Pull request by artek0chumak**
Make possible to use up-to-date PEFT library.

TODO:
- [x] Fix initialization
- [x] Fix inference without adapters
- [x] Fix init for quantized modules
- [x] Fix debug messages while server initialization

### CODE REVIEW

**Review by justheuristic — APPROVED** [review]
LGTM, thank you!

File: src/petals/utils/peft.py [hunk 1]
```diff
@@ -151,67 +153,60 @@ def active_adapter(self):
...
         assert value == self.ADAPTER_NOT_SET, "active adapter can only be changed via .using_adapter" ""
 
+    @property
+    def active_adapters(self):
+        return [self._context_active_adapter]
+
+    def set_adapter(self, adapter_names) -> None:
+        """
+        In PEFT, this function making adapter trainable. However, in Petals environment is not possible now. So,
+        this code remove this functionality.
```
Comment (mryab) [inline src/petals/utils/peft.py:163]:
```suggestion
        In PEFT, this function makes the adapter trainable. However, in Petals environment this is not possible now. Thus,
        this code removes this functionality.
```

File: src/petals/utils/peft.py [hunk 2]
```diff
@@ -151,67 +153,60 @@ def active_adapter(self):
...
+    def __init__(self, base_layer, adapter_name: str):
+        nn.Module.__init__(self)
+        lora.LoraLayer.__init__(self, base_layer)
+
+        self._active_adapter = adapter_name
+        self.is_target_conv_1d_layer = False
+
 
-class LoraLinear8bitLt(AdapterContextMixin, lora.Linear8bitLt):
+# TODO: Check if lora.Linear can be mixed with lora.Linear8bitLt
```
Comment (mryab) [inline src/petals/utils/peft.py:183]: Is this checked?

Comment (artek0chumak) [reply src/petals/utils/peft.py]: Yes, I checked that it worked, and the outputs were the same.

**Pull request merged**